batch2/batch/driver/scheduler.py

```diff
@@ -19,6 +19,33 @@ def __init__(self, app):
         asyncio.ensure_future(self.loop('cancel_loop', self.cancel_state_changed, self.cancel_1))
         asyncio.ensure_future(self.bump_loop())
 
+    async def compute_fair_share(self):
+        free_cores_mcpu = sum([worker.free_cores_mcpu for worker in self.inst_pool.healthy_instances_by_free_cores])
+        total_cores_mcpu = len(self.inst_pool.healthy_instances_by_free_cores) * self.inst_pool.worker_cores * 1000
+        allocated_cores = {}
+
+        records = self.db.execute_and_fetchall(
+            '''
+SELECT user, ready_cores_mcpu, running_cores_mcpu
+FROM user_resources
+WHERE ready_cores_mcpu > 0
+ORDER BY ready_cores_mcpu ASC;
+''')
+
+        n_users = len(records)
+        n_unallocated_users = n_users
+        async for record in records:
+            max_cores_mcpu = (total_cores_mcpu / n_users) - record['running_cores_mcpu'] + \
```

Comment:
So your share isn't (total_cores_mcpu / n_users). If some users need fewer (have fewer ready + running than the even share), you (who have more to run) should get more.

Example: 1000 cores, you have a ton running and I have 5 ready and 5 running, I should get allocated 5 and you should get allocated 995.  Here you'll get at most 500.